learningresources/models.py

```diff
@@ -228,3 +202,46 @@ class Meta:  # pylint: disable=missing-docstring
             RepoPermission.add_edit_metadata,
             RepoPermission.manage_repo_users,
         )
+
+
+def get_preview_url(resource, org=None, course_number=None, run=None):
+    """
+    Create a preview URL. Accepts optional kwargs to prevent
+    database lookups, especially for during search engine indexing.
+    Args:
+        resource (LearningResource): LearningResource
+        org (unicode): resource.course.org
+        run (unicode): resource.course.run
+        course_number (unicode): resource.course.course_number
+    Returns:
+        url (unicode): Preview URL for LearningResource.
+    """
+    if org is None:
+        org = resource.course.org
+    if course_number is None:
+        course_number = resource.course.course_number
+    if run is None:
+        run = resource.course.run
+    key = "{org}/{course}/{run}".format(
+        org=org,
+        course=course_number,
+        run=run,
+    )
+
+    if resource.url_name is not None:
+        url_format = 'courses/{key}/jump_to_id/{preview_id}'
+        return LORE_PREVIEW_BASE_URL + urllib_parse.quote(
```

Comment:
like, I said, feel free to leave it as it is: I'm just going through the "green lines"